x/src/cargo.rs

```diff
@@ -263,6 +304,15 @@ impl<'a> CargoCommand<'a> {
         }
     }
 
+    pub fn get_extra_env(&self) -> &[(&str, &str)] {
+        match self {
+            CargoCommand::Bench(_) => &[],
+            CargoCommand::Check => &[],
+            CargoCommand::Clippy(_) => &[],
+            CargoCommand::Test { env, .. } => &env,
+        }
+    }
+
```

Comment:
I see now the issue you were having with needing to call `to_owned` all the time. There may be a more elegant way to express this (maybe @sunshowers knows a better way) but the following would eliminate the need to alloc:
```suggestion
        pub fn get_extra_env(&self) -> impl IntoIterator<Item = (&str, &str)> {
        let envs: &[(&str, &str)] = match self {
            CargoCommand::Bench(_) => &[],
            CargoCommand::Check => &[],
            CargoCommand::Clippy(_) => &[],
            CargoCommand::Test { env, .. } => &env,
        };
        envs.iter().map(|(k, v)| (*k, *v))
    }


```